File: python/character_match.py
# Don't forget to run the tests (and create some of your own)

# Part 1
def is_character_match(string1, string2):
	match1 = ""
	match2 = ""
	
	for i in string1.lower():
		if i.isalnum() and i in string2.lower():
			match1 += i

	for i in string2.lower():
		if i.isalnum() and i in string1.lower():
			match2 += i
	
	if len(match1) == len(match2):
		return True
	else:
		return False

# Part 2
def anagrams_for(word, list_of_words):
	result = []
	for item in list_of_words:
		matches = ""
		for j in item:
			if j in word:
				matches += j
		if len(matches) == len(word):
			result.append(matches)
	print(result)
	return result

File: js/characterMatch.js
// exports.isCharacterMatch = function(string1, string2) {

// };

// exports.anagramsFor = function(word, listOfWords) {

// };

function isCharacterMatch(string1, string2){
    
    let match1 = ""
    let match2 = ""

    for (const char of string1.toLowerCase()) {
        if (char.match(/[a-z0-9]/) && string2.toLowerCase().includes(char)) {
            match1 += char;
        }
    }
    
    for (const char of string2.toLowerCase()) {
        if (char.match(/[a-z0-9]/) && string1.toLowerCase().includes(char)) {
            match2 += char;
        }
    }

    if (match1.length == match2.length) {
        return true;
    } else {
        return false;
    }
}

function anagramsFor(word, listOfWords) {
    
    let result = []
    
    for (const item of listOfWords) {
        let matches = "";
        for (const char of item) {
            if (word.includes(char)) {
                matches += char
            }
        }
        if (matches.length == word.length) {
            result.push(matches);
        }
    }
    return result;
}

module.exports = { isCharacterMatch, anagramsFor };